### PR TITLE
check quota path syntax acccording to regex, not URI

### PR DESCRIPTION
**Pull request by mvocu**
- The changes concern all quota attributes and methods validating their content.
- The original URI syntax checks allowed for transformations changing the path meaning,
  which is undesirable.

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java [hunk 1]
```diff
@@ -784,11 +785,10 @@ public void checkIfQuotasIsInLimit(Map<String, Pair<BigDecimal, BigDecimal>> quo
...
 			String canonicalPath;
-			try {
-				canonicalPath = new URI(path).normalize().getPath();
-				//path should not end on '/' (problem with some systems as GPFS)
-				if(!canonicalPath.equals("/") && canonicalPath.endsWith("/")) canonicalPath = canonicalPath.substring(0, canonicalPath.length() - 1);
-			} catch (URISyntaxException ex) {
+			canonicalPath = Paths.get(path).normalize().toString();
+			//path should not end on '/' (problem with some systems as GPFS)
+			if(!canonicalPath.equals("/") && canonicalPath.endsWith("/")) canonicalPath = canonicalPath.substring(0, canonicalPath.length() - 1);
+			if(!canonicalPath.matches("^[-a-zA-Z.0-9_/:=,]+$")) {
```
Comment (metodej) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java:791]: The regex could be maybe a constant?

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/ModulesUtilsEntryIntegrationTest.java [hunk 2]
```diff
@@ -715,6 +715,15 @@ public void checkQuotasErrorInValue() throws Exception {
 		modulesUtilsBl.checkAndTransferQuotas(dataQuotasAttribute, resource, null, true);
 	}
 
+	@Test(expected=WrongAttributeValueException.class)
+	public void checkQuotasErrorInPath() throws Exception {
+		System.out.println(CLASS_NAME + "WrongAttributeValueException");
```
Comment (metodej) [inline perun-core/src/test/java/cz/metacentrum/perun/core/entry/ModulesUtilsEntryIntegrationTest.java:720]: I think there is supposed to be name of the test in comment.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java [hunk 3]
```diff
@@ -784,11 +785,10 @@ public void checkIfQuotasIsInLimit(Map<String, Pair<BigDecimal, BigDecimal>> quo
 
 			//testing if path is unique
 			String canonicalPath;
-			try {
-				canonicalPath = new URI(path).normalize().getPath();
-				//path should not end on '/' (problem with some systems as GPFS)
-				if(!canonicalPath.equals("/") && canonicalPath.endsWith("/")) canonicalPath = canonicalPath.substring(0, canonicalPath.length() - 1);
-			} catch (URISyntaxException ex) {
+			canonicalPath = Paths.get(path).normalize().toString();
```
Comment (metodej) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java:788]: I think the declaration and the assignment of the variable could be on one line.

**Review by metodej — APPROVED** [review]

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java (hunk 3 above)
Comment (metodej) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java:788]: Also I think that the `normalize()` function changes slashes in windows and then the regex does not match there. But I don't know if we care about users with windows...

Comment (stavamichal) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java:788]: We are not supporting window's path at this moment so this is ok.

**Review by balcirakpeter — APPROVED** [review]
Please add a short description, why you are making this change (mentioned transformations in the ticket). It may help in the future.
Otherwise ok (when you fix things according to the Metodej's review).

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java (hunk 1 above)
Comment (balcirakpeter) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java:791]: You are right Metodej, but even better would be to create a static variable outside of this method and precompile the regex there so it does not have to be compiled each time this method Is called.

Comment (mvocu) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java:791]: Fixed.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java (hunk 3 above)
Comment (mvocu) [reply perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java:788]: Fixed.

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/ModulesUtilsEntryIntegrationTest.java (hunk 2 above)
Comment (mvocu) [reply perun-core/src/test/java/cz/metacentrum/perun/core/entry/ModulesUtilsEntryIntegrationTest.java:720]: Fixed in the original method also.

**Review by zlamalp — APPROVED** [review]

**Pull request merged**